### PR TITLE
Implemented the search functionality.

### PR DESCRIPTION
**Pull request by Musa-Mabasa**
This branch represents the whole search functionality with the help of rxjs and tanstack query. 

N.B : If you want to run the app just note that it's not responsive yet 🔧

### CODE REVIEW

File: src/features/search/components/QuickSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import AdvancedSearch from "./AdvancedSearch";
+
+const QuickSearch = () => {
+  return (
+    <div className="grow py-32">
+      <h1 className="text-lg font-semibold pt-5  px-4">Quick Search</h1>
+      <AdvancedSearch />
```
Comment (tanadendere) [inline src/features/search/components/QuickSearch.tsx]: Is this quick search's functionality only to call the advanced search?

File: src/features/search/components/Results.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      {comics && comics.length ? (
+        <div className="flex flex-col">
+          <div className="py-32 px-4 grid grid-cols-2 gap-y-8 place-items-center grow-0 overflow-auto bg-gradient-to-t from-primary-color to-gradient-color">
+            {comics?.map((comic) => (
+              <MangaCard key={comic?.comic?.slug} cardInfo={comic} />
+            ))}
+          </div>
+        </div>
+      ) : (
+        <h1>Nothing</h1>
```
Comment (tanadendere) [inline src/features/search/components/Results.tsx]: I would suggest changing this to a more verbose user feedback. Like "No results found for your search..."

File: src/features/search/components/Results.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  const loadingQuery = comicQueries?.find((query) => query.isPending);
+  const errorQuery = comicQueries?.find((query) => query.isError);
+
+  if (isLoading || loadingQuery) {
+    return <ResultSkeleton />;
+  }
+
+  if (errorQuery) {
+    return (
+      <div className="h-full flex items-center justify-center"> No manga</div>
```
Comment (tanadendere) [inline src/features/search/components/Results.tsx:56]: Same this here as well about the user feedback

File: src/features/search/components/AdvancedSearch.tsx [hunk 4]
```diff
@@ -1,27 +1,96 @@
...
+          return { ...listStatus, isChecked: false };
+        }
+      })
+    );
+  }
+
+  async function handleSearch() {
+    queryGenres.current = genres.filter((genre) => genre.isChecked).join(",");
+
+    for (let status of statuses) {
```
Comment (Julianna-Venter) [inline src/features/search/components/AdvancedSearch.tsx]: can make `let` a `const` instead

File: src/features/search/components/SearchInput.tsx [hunk 5]
```diff
@@ -1,16 +1,113 @@
...
+          }
+          return (event.target as HTMLInputElement).value;
+        }),
+        debounceTime(600),
+        distinctUntilChanged(),
+        switchMap((value) => {
+          if (value !== "") {
+            setisLoading(true);
+            return fromFetch(
+              `${apiBasieURL}/v1.0/search/?page=1&limit=15&showall=false&q=${value}&t=false`
```
Comment (Julianna-Venter) [inline src/features/search/components/SearchInput.tsx:44]: Why did you use useEffect & Observable over useQuery?

**Review by Julianna-Venter — APPROVED** [review]
One small change, and just make sure you can support your usage of useEffect over useQuery.

File: src/features/search/components/QuickSearch.tsx (hunk 1 above)
Comment (Musa-Mabasa) [reply src/features/search/components/QuickSearch.tsx:7]: Thank you. I forgot about that.

File: src/features/search/components/AdvancedSearch.tsx (hunk 4 above)
Comment (Musa-Mabasa) [reply src/features/search/components/AdvancedSearch.tsx]: Oh yes, thank you.

File: src/features/search/components/QuickSearch.tsx (hunk 1 above)
Comment (Musa-Mabasa) [reply src/features/search/components/QuickSearch.tsx]: I will make changes on that, thank you.

File: src/features/search/components/Results.tsx (hunk 3 above)
Comment (Musa-Mabasa) [reply src/features/search/components/Results.tsx:56]: For now, these are place holders, I will add more coherent feedback in a different branch.

File: src/features/search/components/SearchInput.tsx (hunk 5 above)
Comment (Musa-Mabasa) [reply src/features/search/components/SearchInput.tsx:44]: Observables were the quickest way I could get the debounce functionality working without racking my brain too much on something new and unfamiliar.

**Review by tanadendere — APPROVED** [review]
All comments were resolved

**Pull request merged**